travis/apis.py

```diff
@@ -122,6 +123,19 @@ def _component_lock(self, component, lock=True):
             time.sleep(60)
         return True
 
+    def _component_commit(self, component):
+        url = (self.host + '/components/%s/%s/repository/' %
+               (self.project['slug'], component['slug']))
+        needs_commit = self._request(url)
+        if not needs_commit['needs_commit']:
+            return
+        for i in range(10):
```

Comment:
Yeah, but iterating several times is not going to be the solution. If the commit can't be done, the result will be the same after 10 times. I know that you think in connection problems, but check this, not a general try.